### PR TITLE
[generator][python] Fixed files copy.

### PR DESCRIPTION
**Pull request by vng**

### CODE REVIEW

File: tools/python/maps_generator/utils/file.py [hunk 1]
```diff
@@ -187,9 +187,22 @@ def is_verified(name: AnyStr) -> bool:
 
 
 def copy_overwrite(from_path: AnyStr, to_path: AnyStr):
-    if os.path.exists(to_path):
-        shutil.rmtree(to_path)
-    shutil.copytree(from_path, to_path)
+    try:
+        # Workaround: shutil.copytree/rmtree (3.10) fails if src is a symlink or file ?!
```
Comment (biodranik) [inline tools/python/maps_generator/utils/file.py:191]: Is it a guess? What is the exact issue and what causes it?

Comment (vng) [reply tools/python/maps_generator/utils/file.py:191]:
```
Python 3.11.3 (main, Apr  7 2023, 20:13:31) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import shutil
>>> shutil.copytree("./omim_config.h", "./tmp.h")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/shutil.py", line 559, in copytree
    with os.scandir(src) as itr:
         ^^^^^^^^^^^^^^^
NotADirectoryError: [Errno 20] Not a directory: './omim_config.h'
```

Comment (biodranik) [reply tools/python/maps_generator/utils/file.py:191]: 1. [shutil.copytree documentation](https://docs.python.org/3/library/shutil.html#shutil.copytree) clearly states that it works with directories, not with files. Trying to copy a file or copy a dir into a file with this method should fail.
2. It copies directory symlinks without errors.

I found only one place where `copy_overwrite()` it is used:
```
    if is_file_uri(url):
        # url uses 'file://' scheme
        copy_overwrite(file_uri_to_path(url), name)
        logger.info(f"File {name} was copied from {url}.")
        return
```
Here a _file_ is copied, not a directory. Why `shutil.copy2` can't be used there directly, and `copy_overwrite` can't be removed?

Comment (vng) [reply tools/python/maps_generator/utils/file.py:191]: - I didn't write this code
- It calls rm first

Comment (biodranik) [reply tools/python/maps_generator/utils/file.py:191]: copy2 overwrites destination if it exists. There is no need for this patch, just remove the code and use copy2 instead.

**Comment by vng** [comment]
PTAL

**Review by biodranik — APPROVED** [review]

**Comment by koenvanhollebeke** [comment]
fails if the directory where the file is copied to does not exist

> [2023-05-21 08:34:32,503] INFO stages_declaration Mwm stage StageRoutingTransit: start...
[2023-05-21 08:34:32,516] INFO file Trying to download /home/koenvh/maps_build/2023_05_21__08_29_31/intermediate_data/transit_from_gtfs/gates.transit.json from file:///home/koenvh/Downloads/Indonesia_Jawa Barat/gates.transit.json.
> [2023-05-21 08:34:32,541] ERROR env Uncaught exception: [Errno 2] No such file or directory: '/home/koenvh/maps_build/2023_05_21__08_29_31/intermediate_data/transit_from_gtfs/gates.transit.json'

> File "/usr/lib/python3.10/shutil.py", line 256, in copyfile
    with open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: '/home/koenvh/maps_build/2023_05_21__08_29_31/intermediate_data/transit_from_gtfs/gates.transit.json'

if the destination directory does not exist it needs to be created first.
Python 3.10.6/Ubuntu 22.04LTS

**Pull request merged**